### DISCUSSION TITLE
weird problem while creating transcriptome index using salmon

### DISCUSSION THREAD

**Discussion by bzmby** [nextflow-io/nextflow#2597] [Q&A] [answered]
I am trying to m make transcriptome index using salmon and the following script:  

```
#!/usr/bin/env nextflow
nextflow.preview.dsl=2

params.transcriptome = "/hg38_genome/gencode.v39.transcripts.fa"
params.outdir = "results"


process INDEX {

    input:
    path transcriptome

    output:
    path 'index'

    script:
    """
    /common/Salmon/0.13.1/salmon-latest_linux_x86_64/bin/salmon index --thre$
    """
}

transcriptome_ch = channel.fromPath(params.transcriptome)

workflow {
   INDEX(transcriptome_ch)
}
```

I used this shell command to run it:

    nextflow run script.nf

But I got this error:

    No such variable: channel
    
     -- Check script 'script.nf' at line: 38 or see '.nextflow.log' file for more details

Do you know how I can fix the issue?
I am using cextflow version 20.04.1.

**Comment by pditommaso** [comment, marked as answer]
Update nextflow to latest version
